### PR TITLE
e2e: validate subResourceNamePrefix is used to name appgw sub-resources

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR adds
1. fixes a bug in the placement of subResourceNamePrefix in the helm config.
2. an e2e that uses subResourceNamePrefix in the helm and validates that app gateway sub-resources are named apt-ly.

### CODE REVIEW

File: scripts/e2e/cmd/runner/environment.go [hunk 1]
```diff
@@ -28,25 +28,35 @@ const (
 
 	// ObjectIDVarName is tne name of the identityObjectId
 	ObjectIDVarName = "identityObjectId"
+
+	// SubResourceNamePrefixVarName is tne name of the subResourceNamePrefix
+	SubResourceNamePrefixVarName = "subResourceNamePrefix"
+
+	// AzureAuthLocationVarName is tne name of the AZURE_AUTH_LOCATION
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/environment.go:35]: nit: typo "tne"

File: scripts/e2e/cmd/runner/helper.go [hunk 2]
```diff
@@ -397,3 +434,45 @@ func makeGetRequest(url string, host string, statusCode int, inSecure bool) (*ht
...
+		return err
+	}
+
+	prefixUsed := false
+	for _, listener := range *gateway.HTTPListeners {
+		klog.Infof("checking listener %s for %s", *listener.Name, env.SubResourceNamePrefix)
+		if strings.HasPrefix(*listener.Name, env.SubResourceNamePrefix) {
+			klog.Infof("found %s that uses the prefix", *listener.Name)
+			prefixUsed = true
+			break
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/helper.go:468]: what if other listener doesn't have the prefix? prefixUsed should be initialized to be true and check if anything doesn't have the prefix.

File: scripts/e2e/cmd/runner/helper.go [hunk 3]
```diff
@@ -397,3 +434,45 @@ func makeGetRequest(url string, host string, statusCode int, inSecure bool) (*ht
 
 	return nil, fmt.Errorf("Unable to get status code %d with url '%s' with host '%s'. Response: [%+v]", statusCode, url, host, resp)
 }
+
+func validateSubResourcePrefix() error {
+	klog.Info("Validating that appgw sub resources are named with the prefix provided in helm")
+
+	env := GetEnv()
+	if env.SubResourceNamePrefix == "" {
+		return errors.New("value for SubResourceNamePrefix env variable is not set and so, can't validate that all appgw properties use the prefix")
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/helper.go:443]: typo: "so can't"?

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go [hunk 4]
```diff
@@ -131,7 +132,7 @@ var _ = Describe("MFU", func() {
 
 		It("[retry access check] should be able to wait for the access to be granted", func() {
 			klog.Info("Initializing role client")
-			roleClient, err := getRoleClient()
+			roleClient, err := getRoleAssignmentsClient()
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:135]: unrelated changes go to different PR?

File: scripts/e2e/cmd/runner/helper.go (hunk 2 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/helper.go:468]: since not all resources might use the prefix, just checking that prefix is getting to AGIC.
Cases where this would fail is when AppGW is shared or ssl certificates are pre-installed through portal.

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go (hunk 4 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:135]: sure

File: scripts/e2e/cmd/runner/lfu_one_namespace_one_ingress_test.go [hunk 5]
```diff
@@ -52,6 +55,47 @@ var _ = Describe("LFU", func() {
...
+			SSLE2ERedirectYamlPath := "testdata/one-namespace-one-ingress/ssl-e2e-redirect/app.yaml"
+			klog.Info("Applying yaml: ", SSLE2ERedirectYamlPath)
+			err = applyYaml(clientset, namespaceName, SSLE2ERedirectYamlPath)
+			Expect(err).To(BeNil())
+			time.Sleep(30 * time.Second)
+
+			gateway, err := getGateway()
+			Expect(err).To(BeNil())
+
+			prefixUsed := false
```
Comment (3quanfeng) [inline scripts/e2e/cmd/runner/lfu_one_namespace_one_ingress_test.go:83]: if all the listeners are supposed to be prefixed, then it makes more sense to check if any listener doesn't have the prefix

**Review by 3quanfeng — APPROVED** [review]
LGTM!
